For giac 1.5.0-37. Fix an overflow in gbasis computations modulo primes between 2**30 and 2**31
cf: https://trac.sagemath.org/ticket/27306

using upstream patch from:
https://dev.geogebra.org/trac/changeset/66972/

Index: /trunk/geogebra/giac/src/giac/cpp/cocoa.cc
===================================================================
--- a/src/cocoa.cc	(revision 66971)
+++ b/src/cocoa.cc	(revision 66972)
@@ -12619,8 +12619,8 @@
       n %= env;
 #endif
-      if (n*2>env)
+      if (n*2LL>env)
 	it->g -= env;
       else {
-	if (n*2<=-env)
+	if (n*2LL<=-env)
 	  it->g += env;
       }
